vkcs/backup/backup_shared.go

```diff
@@ -0,0 +1,273 @@
+					break
+				}
+			}
+		}
+		if len(resourcesInfo) != len(instancesID) {
+			return nil, fmt.Errorf("error getting resources info: some resources were not found")
+		}
+
+		return resourcesInfo, nil
+	}
+
+	if resourceType == ProviderNameTrove {
+		dbClient, err := config.DatabaseV1Client(region)
+		if err != nil {
+			return nil, fmt.Errorf("error creating VKCS database client: %s", err.Error())
+		}
+
+		allPages, err := instances.List(dbClient).AllPages()
+		if err != nil {
+			return nil, fmt.Errorf("error getting database instances info: %s", err)
+		}
+
+		allInstances, err := instances.ExtractInstances(allPages)
+		if err != nil {
+			return nil, fmt.Errorf("error getting database instances info: %s", err)
+		}
+
+		resourcesInfo := make([]*plans.BackupPlanResource, 0)
+		for _, instanceID := range instancesID {
+			for _, dbInstance := range allInstances {
+				if instanceID.ValueString() == dbInstance.ID {
+					resourceInfo := plans.BackupPlanResource{
+						ID:   dbInstance.ID,
+						Type: TroveInstance,
+						Name: dbInstance.Name,
+					}
+					resourcesInfo = append(resourcesInfo, &resourceInfo)
+					break
+				}
+			}
+		}
+		if len(resourcesInfo) != len(instancesID) {
+			return nil, fmt.Errorf("error getting resources info: some resources were not found")
+		}
+	}
+	return nil, fmt.Errorf("error getting resources info: unknown resource type")
+}
+
+func dayToNumber(day string) int {
+	days := map[string]int{
+		"Mo": 0,
+		"Tu": 1,
+		"We": 2,
+		"Th": 3,
+		"Fr": 4,
+		"Sa": 5,
+		"Su": 6,
+	}
+	return days[day]
+}
+
+func numberToDay(number int) string {
+	days := []string{"Mo", "Tu", "We", "Th", "Fr", "Sa", "Su"}
+	return days[number]
+}
+
+func expandFullDay(plan PlanResourceModel) (int, error) {
+	scheduleDates := plan.Schedule.Date
+	if len(scheduleDates) > 1 {
+		return 0, fmt.Errorf("invalid resource schema. There should be only one date for incremental_backups")
+	}
+	fullDay := dayToNumber(scheduleDates[0].ValueString())
+	return fullDay, nil
+}
+
+func expandGFS(plan PlanResourceModel) *plans.BackupPlanGFS {
+	gfs := &plans.BackupPlanGFS{
+		Son: int(plan.GFSRetention.GFSWeekly.ValueInt64()),
+	}
+	if !plan.GFSRetention.GFSMonthly.IsNull() {
+		gfs.Father = int(plan.GFSRetention.GFSMonthly.ValueInt64())
+	}
+	if !plan.GFSRetention.GFSYearly.IsNull() {
+		gfs.Grandfather = int(plan.GFSRetention.GFSYearly.ValueInt64())
+	}
+	return gfs
+}
+
+func expandTriggerSchedule(plan PlanResourceModel) (string, error) {
+	var triggerSchedule string
+	if !plan.Schedule.EveryHours.IsNull() {
+		everyHours := plan.Schedule.EveryHours.ValueInt64()
+		triggerSchedule = fmt.Sprintf("0 */%d * * *", everyHours)
+	} else {
+		incrementalBackups := plan.IncrementalBackup.ValueBool()
+		if incrementalBackups {
+			planTime, err := time.Parse(TimeFormat, plan.Schedule.Time.ValueString())
+			if err != nil {
+				return "", fmt.Errorf("invalid resource schema. Invalid time")
+			}
+			planTime = planTime.Add(-time.Hour * 3)
```

Comment:
GMT?

Reply:
Yes, user sets time as GMT+3, same as on frontend, then we convert it. Added timezone info to schedule.time argument description

Reply:
Changed to UTC+0 timezone